subsys/bluetooth/host/hci_raw.c

```diff
@@ -62,26 +63,12 @@ struct net_buf *bt_buf_get_rx(enum bt_buf_type type, s32_t timeout)
 
 struct net_buf *bt_buf_get_cmd_complete(s32_t timeout)
 {
-	struct net_buf *buf;
-
-	buf = net_buf_alloc(&hci_rx_pool, timeout);
-	if (buf) {
-		bt_buf_set_type(buf, BT_BUF_EVT);
-	}
-
-	return buf;
+	return bt_buf_get_rx(BT_BUF_EVT, timeout);
```

Comment:
@jhedberg bt_buf_get_cmd_complete seem to only be called internally by hci_core.c through `bt_buf_get_evt`.
Should this perhaps be declared as an internal function? And everyone should go through `bt_buf_get_rx`?

Reply:
`bt_buf_get_cmd_complete()` has been deprecated (just forgotten to be annotated like that) which is why no-one calls it externally anymore. `bt_buf_get_rx()` doesn't work for everything since it doesn't get the HCI event code as a parameter (which would be needed for the "command complete/status" special case). That said, I'm sure there's room for improving this API - I'd be happy to consider proposals for Zephyr 2.2.